kingpin/actors/aws/cloudformation.py

```diff
@@ -0,0 +1,434 @@
+
+    """Creates an Amazon CF Stack.
+
+    http://boto.readthedocs.org/en/latest/ref/cloudformation.html
+    #boto.cloudformation.connection.CloudFormationConnection.create_stack
+
+    """
+
+    all_options = {
+        'capabilities': (list, [],
+                         'The list of capabilities that you want to allow '
+                         'in the stack'),
+        'disable_rollback': (bool, False,
+                             'Set to `True` to disable rollback of the stack '
+                             'if stack creation failed.'),
+        'name': (str, REQUIRED, 'Name of the stack'),
+        'parameters': (dict, {}, 'Parameters passed into the CF '
+                                 'template execution'),
+        'region': (str, REQUIRED, 'AWS region name, like us-west-2'),
+        'template': (str, REQUIRED,
+                     'Path to the AWS CloudFormation File. http(s)://, '
+                     'file:///, absolute or relative file paths.'),
+        'timeout_in_minutes': (int, 60,
+                               'The amount of time that can pass before the '
+                               'stack status becomes CREATE_FAILED'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Initialize our object variables."""
+        super(Create, self).__init__(*args, **kwargs)
+
+        # Check if the supplied CF template is a local file. If it is, read it
+        # into memory.
+        (self._template_body, self._template_url) = self._get_template_body(
+            self.option('template'))
+
+    def _get_template_body(self, template):
+        """Reads in a local template file and returns the contents.
+
+        If the template string supplied is a local file resource (has no
+        URI prefix), then this method will return the contents of the file.
+        Otherwise, returns None.
+
+        Args:
+            template: String with a reference to a template location.
+
+        Returns:
+            Tuple with:
+              (None/Contents of template file,
+               None/URL of template)
+
+        Raises:
+            InvalidTemplate
+        """
+        remote_types = ('http://', 'https://')
+
+        if self.option('template').startswith(remote_types):
+            return (None, template)
+
+        try:
+            return (open(template, 'r').read(), None)
+        except IOError as e:
+            raise InvalidTemplate(e)
+
+    @concurrent.run_on_executor
+    @retry(retry_on_exception=retry_if_transient_error,
+           stop_max_attempt_number=5,
+           wait_exponential_multiplier=500,
+           wait_exponential_max=aws_settings.CF_WAIT_MAX)
+    @utils.exception_logger
+    def _validate_template(self):
+        """Validates the CloudFormation template.
+
+        Raises:
+            InvalidTemplate
+            exceptions.InvalidCredentials
+        """
+        if self._template_body is not None:
+            self.log.debug('Validating template with AWS...')
```

Comment:
I think these two should be `.info()` -- since it may take a second to check it would be valuable to see it.